src/Arbor/Api/Gateway/RestGateway.php

```diff
@@ -646,7 +646,7 @@ public function sendRequest($method, $url, array $options = [])
             throw new ServerErrorException('An unexpected error has occurred: ' . $e->getMessage(), 0, $e);
         }
 
-        if (!is_array($responsePayload)) {
+        if (!is_array($responsePayload) && $code !== 204) {
```

Comment:
Instead of checking the response code here - we could just change line 682 to be return `$responsePayload ?? [];` because is_array would then return true with an empty payload. Just an idea @SaleStojkovic

Reply:
Yeah, but exception will be thrown here. Not sure if we could remove this `is_array` check completely. What do you think @arborluke ?

Reply:
Not if `$responsePayload` is an array? I dont think we can make that sort of call together. Maybe a decision for @igorvuckovic but I think this looks OK as it is?

Reply:
It is array. I also think this looks ok.

Reply:
I think we can remove the status code check   `&& $code !== 204` - responsePayload will be an array for a 204 due to the change below @SaleStojkovic

Reply:
No, it won't. I've tested locally. The problem is, as I said on Slack, delete api returns empty payload with 204 status code, so when calling delete api this if will be true without code check and exception will be thrown.

Reply:
OK, @igorvuckovic I wonder if we can remove this check for an array. Happy to approve in its current state but maybe we can do one better.